src/test/java/com/wikia/webdriver/common/logging/PageObjectLogging.java

```diff
@@ -129,22 +133,58 @@ public static void logOnLowLevel(String command, String description, boolean suc
     log(command, description, success, true);
   }
 
+  //TODO: Check error class here
   private static void log(String command, String description, boolean isSuccess,
-      boolean ifLowLevel) {
+                   boolean ifLowLevel) {
     logsResults.add(isSuccess);
     String escapedDescription = escapeHtml(description);
 
-    String className = isSuccess ? "success" : "error";
-    StringBuilder builder = new StringBuilder();
-    if (ifLowLevel) {
-      builder.append("<tr class=\"" + className + " lowLevelAction" + "\"><td>" + command
-          + "</td><td>" + escapedDescription + "</td><td> <br/> &nbsp;</td></tr>");
-    } else {
-      builder.append("<tr class=\"" + className + "\"><td>" + command + "</td><td>"
-          + escapedDescription + "</td><td> <br/> &nbsp;</td></tr>");
+    String className = isSuccess ? SUCCESS_CLASS : ERROR_CLASS;
+//    if(!isSuccess){
+//      lastThrow = new Throwable();
+//      lastThrow.setStackTrace(Thread.currentThread().getStackTrace());
+//    }
+    List<String> classList = new ArrayList<>();
+    classList.add(className);
+
+    if (ifLowLevel)
+      classList.add(LOW_LEVEL_ACTION_CLASS);
+      String html = VelocityWrapper.appendLogRow(classList, command, escapedDescription);
+      CommonUtils.appendTextToFile(logPath, html);
+      logJSError(DriverProvider.getActiveDriver());
+
+      if (!isSuccess) {
+        lastThrow = new Throwable();
+        lastThrow.setStackTrace(Thread.currentThread().getStackTrace());
+//      logStacktrace();
+      }
+  }
+
+  public void logAssertionStacktrace(AssertionError exception) {
+    driver = DriverProvider.getActiveDriver();
+
+    imageCounter += 1;
+    if ("true".equals(Configuration.getLogEnabled())) {
+      try {
+        new Shooter().savePageScreenshot(screenPath + imageCounter, driver);
+        CommonUtils.appendTextToFile(screenPath + imageCounter + ".html", getPageSource(driver));
+      } catch (Exception e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)